src/js/planting/planting-handlers.js

```diff
@@ -93,33 +88,76 @@ Planting.prototype.plant_objects_for_view = function () {
             that.plantedobjects[i].img = img;
             that.plantedobjects[i].container = div;
         }
+    };
+};
+
+var calc_direction = function calc_direction(objectPosition, mousePosition) {
+    var calc = {};
+
+    if (objectPosition < mousePosition) {
+        calc.direction = "ASC";
+        calc.width = mousePosition - objectPosition;
+    } else if (objectPosition > mousePosition) {
+        calc.direction = "DSC";
+        calc.width = objectPosition - mousePosition;
     }
+
+    return calc;
 };
 
+var isDown = false;
 Planting.prototype.rotate_object = function (e) {
+    isDown = true;
+    $('body').addClass('noselect rotate');
+
     var plantedObject = e.data.plantedObject;
-    var direction = e.data.direction;
-    var toolBoxObjects = e.data.toolBoxObjects;
+    var projections = e.data.toolBoxObjects[plantedObject.object].projections;
 
-    var projections = toolBoxObjects[plantedObject.object].projections;
+    var side_width = 20;
+    var projection;
 
-    if (direction === 'left') {
-        if (plantedObject.projection === 0) {
-            plantedObject.projection = projections.length;
-        } else {
-            plantedObject.projection -= 1;
+    buttonX = $(this).offset().left;
+    currentProjection = plantedObject.projection;
+
+    var calc_projection = function calc_projection(objectPosition, mousePosition) {
+        var rotate = calc_direction(objectPosition, mousePosition);
+        rotate.current = currentProjection;
+        rotate.current++;
+        rotate.sides = (rotate.width - (rotate.width % side_width)) / side_width;
+        if (rotate.sides >= projections.length) {
+            rotate.sides = rotate.sides % projections.length;
         }
-        plantedObject.img = plantedObject.img.attr('src', projections[plantedObject.projection]);
-    } else if (direction === 'right') {
-        if (plantedObject.projection === projections.length) {
-            plantedObject.projection = 0;
-        } else {
-            plantedObject.projection += 1;
+        if (rotate.direction === "ASC") {
+            if ((rotate.current + rotate.sides) > projections.length) {
+                rotate.projection = (rotate.current + rotate.sides) % projections.length;
+            } else {
+                rotate.projection = rotate.current + rotate.sides;
+            }
+        } else if (rotate.direction === "DSC") {
+            if (rotate.current <= rotate.sides) {
+                rotate.current += projections.length;
+            }
+            rotate.projection = rotate.current - rotate.sides;
```

Comment:
Piękna ifologia :)

Ogólnie jak się takie coś zaczyna robić, to znak że trzeba już rozbijać funkcję na mniejsze funkcje i zacząć takie pomniejsze ify pakować właśnie w mniejsze funkcje, które nazwą opisują co się tam dzieje i co zwracają.

Kilka źródeł, które mogą pomóc:
http://blog.codinghorror.com/flattening-arrow-code/
https://javascriptweblog.wordpress.com/2010/07/26/no-more-ifs-alternatives-to-statement-branching-in-javascript/
http://toddmotto.com/deprecating-the-switch-statement-for-object-literals/

Ale to tak jako ciekawostka, teraz już niech to zostanie. Przyjdzie czas na refaktor kodu i upiększanie (mam nadzieję), a póki co ważne żeby w ogóle coś działało.